File: c/grains/src/grains.c
#include "grains.h"
#include <math.h>
#include <stdint.h>
#include <stdio.h>

#define MAXSQR 64

uint64_t total(void) { return UINT64_MAX; }

uint64_t square(uint8_t index) {
  return (index > MAXSQR || index <= 0) ? 0 : (uint64_t)1 << (index - 1);
}

File: go/grains/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(square int) (uint64, error) {
	if square <= 0 || square > 64 {
		return 0, errors.New("square must be between 1-64")
	}
	return uint64(math.Pow(2, float64(square-1))), nil
}

func Total() uint64 {
	return ^uint64(0)
}
